activesupport/lib/active_support/number_helper/number_to_rounded_converter.rb

```diff
@@ -30,7 +30,7 @@ def convert
           formatted_string =
             if BigDecimal === rounded_number && rounded_number.finite?
               s = rounded_number.to_s('F') + '0'*precision
-              a, b = s.split('.', 2)
+              a, b = s.split('.'.freeze, 2)
               a + '.' + b[0, precision]
```

Comment:
Weird, this gives you an error

```
'.'.freeze << "what"
RuntimeError: can't modify frozen String
```

I'm surprised that putting another string in front doesn't. I guess this is because `a << '.'.freeze` gets evaluated first and returns a non-frozen string.

I would have done the more verbose

```ruby
a << '.'.freeze
a << b[0, precision]
```

Your way is fine.

Since "s" is also not an input string and it doesn't look like it's used anywhere else we could do the same there

```ruby
s = rounded_number.to_s('F'.freeze)
s << '0'.freeze * precision
```

That would save us two string allocations including getting rid of even more `"0"` strings.